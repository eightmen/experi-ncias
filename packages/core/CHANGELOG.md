# v0.13.0 (Mon Dec 27 2021)

### Release Notes

#### Move Emotion and MDX packages to peerDeps ([#1867](https://github.com/system-ui/theme-ui/pull/1867))

**BREAKING CHANGE:** `@emotion/react` and `@mdx-js/react`  were moved to peerDependencies of `theme-ui`, `@theme-ui/mdx` and `@theme-ui/sidenav` and must be installed separately. This is a revert of change from Theme UI v0.3, and it's meant to help solve version clashes and context mismatch bugs on user side. Connected issues: [#1793](https://github.com/system-ui/theme-ui/issues/1793), [#1531](https://github.com/system-ui/theme-ui/issues/1531), [#1530](https://github.com/system-ui/theme-ui/issues/1530), [#1388](https://github.com/system-ui/theme-ui/issues/1388), [#1345](https://github.com/system-ui/theme-ui/issues/1345#issuecomment-742225675), [#1130](https://github.com/system-ui/theme-ui/issues/1130).

---

---

#### 🚀 Enhancement

- Move Emotion and MDX packages to peerDeps [#1867](https://github.com/system-ui/theme-ui/pull/1867) ([@hasparus](https://github.com/hasparus))
- feat(deps): move Emotion to peerDeps ([@hasparus](https://github.com/hasparus))

#### 🐛 Bug Fix

- fix: add missing deepmerge dependency ([@hasparus](https://github.com/hasparus))

#### 👨‍💻 Minor changes

- docs: Write migration notes for v0.13 ([@hasparus](https://github.com/hasparus))

#### Authors: 1

- Piotr Monwid-Olechnowicz ([@hasparus](https://github.com/hasparus))

---

# v0.12.0 (Thu Oct 28 2021)

:tada: This release contains work from a new contributor! :tada:

Thank you, William Pei ([@draekien](https://github.com/draekien)), for all your work!

#### 🚀 Enhancement


#### 🐛 Bug Fix

- Merge remote-tracking branch 'upstream/develop' into fix/color-scheme-media-query-api-update [#981](https://github.com/system-ui/theme-ui/pull/981) ([@draekien](https://github.com/draekien))

#### Authors: 6

- [@dependabot[bot]](https://github.com/dependabot[bot])
- Julian Benegas ([@julianbenegas](https://github.com/julianbenegas))
- Lachlan Campbell ([@lachlanjc](https://github.com/lachlanjc))
- Lennart ([@LekoArts](https://github.com/LekoArts))
- Piotr Monwid-Olechnowicz ([@hasparus](https://github.com/hasparus))
- William Pei ([@draekien](https://github.com/draekien))

---

# v0.11.0 (Wed Aug 25 2021)

#### 🚀 Enhancement

- fix(color-modes): combine colors in nested providers [#1838](https://github.com/system-ui/theme-ui/pull/1838) ([@hasparus](https://github.com/hasparus))

#### 🐛 Bug Fix

- add types for css prop back [#1866](https://github.com/system-ui/theme-ui/pull/1866) ([@hasparus](https://github.com/hasparus))
- fix(core): add `css` prop back to JSX types ([@hasparus](https://github.com/hasparus))
- 0.10 chores [#1842](https://github.com/system-ui/theme-ui/pull/1842) ([@hasparus](https://github.com/hasparus))
- fix(color-modes): properly combine colors from nested providers ([@hasparus](https://github.com/hasparus))

#### 🏠 Internal

- docs(css): add usage examples for core packages [#1877](https://github.com/system-ui/theme-ui/pull/1877) ([@hasparus](https://github.com/hasparus))

#### Authors: 1

- Piotr Monwid-Olechnowicz ([@hasparus](https://github.com/hasparus))

---

# v0.10.0 (Sat Jun 19 2021)

#### 🏠 Internal

- Docs: update JSX pragma guide with automatic runtime section [#1718](https://github.com/system-ui/theme-ui/pull/1718) ([@flo-sch](https://github.com/flo-sch) [@hasparus](https://github.com/hasparus))

#### Authors: 2

- Florent SCHILDKNECHT ([@flo-sch](https://github.com/flo-sch))
- Piotr Monwid-Olechnowicz ([@hasparus](https://github.com/hasparus))

---

# v0.8.3 (Wed May 05 2021)

#### ⚠️ Pushed to `stable`

- Merge branch 'stable' into develop ([@hasparus](https://github.com/hasparus))

#### Authors: 1

- Piotr Monwid-Olechnowicz ([@hasparus](https://github.com/hasparus))

---

# v0.8.1 (Wed May 05 2021)

#### 🐛 Bug Fix

- Wrap top level Them