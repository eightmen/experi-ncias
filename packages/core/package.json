
{
  "name": "@theme-ui/core",
  "version": "0.15.6-develop.0",
  "source": "src/index.ts",
  "main": "dist/theme-ui-core.cjs.js",
  "module": "dist/theme-ui-core.esm.js",
  "types": "dist/theme-ui-core.cjs.d.ts",
  "sideEffects": false,
  "scripts": {},
  "repository": "system-ui/theme-ui",
  "author": "Brent Jackson",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "@theme-ui/css": "workspace:^",
    "deepmerge": "^4.2.2"
  },
  "peerDependencies": {
    "@emotion/react": "^11",
    "react": ">=18"
  },
  "devDependencies": {
    "@types/react": "^18",
    "@babel/core": "^7",
    "@theme-ui/test-utils": "workspace:^"
  },
  "preconstruct": {
    "entrypoints": [
      "index.ts",
      "jsx-dev-runtime.ts",
      "jsx-runtime.ts"
    ]
  }
}