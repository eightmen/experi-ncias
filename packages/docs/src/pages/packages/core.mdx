---
title: '@theme-ui/core'
---

import Readme from '@theme-ui/core/README.md'

<Readme />
