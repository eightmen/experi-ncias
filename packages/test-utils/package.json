
{
  "name": "@theme-ui/test-utils",
  "version": "0.15.6-develop.0",
  "private": true,
  "license": "MIT",
  "repository": "system-ui/theme-ui",
  "source": "src/index.ts",
  "main": "dist/theme-ui-test-utils.cjs.js",
  "module": "dist/theme-ui-test-utils.esm.js",
  "types": "dist/theme-ui-test-utils.cjs.d.ts",
  "sideEffects": false,
  "scripts": {
    "build": "tsc",
    "watch": "tsc -w",
    "typecheck": "tsc --noEmit"
  },
  "dependencies": {
    "@testing-library/react": "^13.3.0",
    "conditional-type-checks": "^1.0.6",
    "react-test-renderer": "^18.2.0",
    "ts-snippet": "^5.0.0"
  },
  "peerDependencies": {
    "react": ">=18",
    "theme-ui": "workspace:^"
  },
  "devDependencies": {
    "@types/react": "^18",
    "react": "^18",
    "react-dom": ">=18",
    "typescript": ">=4",
    "theme-ui": "workspace:^"
  }
}