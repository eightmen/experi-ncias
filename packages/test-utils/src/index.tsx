import * as React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import * as tsSnippet from 'ts-snippet'
import { Theme, ThemeProvider } from 'theme-ui'

export * from '@testing-library/react'

export type {
  Assert,
  AssertFalse,
  AssertTrue,
  Has,
  IsAny,
  IsExact,
  IsNever,
  IsNullable,
  IsUnknown,
  NotHas,
} from 'conditional-type-checks'

export const renderHook = <T,>(
  useHook: () => T,
  options: { theme?: Theme } = {}
): {
  unmount: () => void
  rerender: (
    ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>
  ) => void
  readonly result: NonNullable<T>
} => {
  let value: T | undefined = undefined

  const Component = () => {
    value = useHook()
    return null
  }

  const { theme } = options
  const { unmount, rerender } = render(<Component />, {
    wrapper: theme && ((props) => <ThemeProvider theme={theme} {...props} />),
  })

  return {
    unmount,
    rerender,
    get result() {
      return value!
    },
  }
}

export const renderJSON = (
  el: Parameters<typeof renderer.create>[0]
): renderer.ReactTestRendererJSON | null => {
  const json = renderer.create(el).toJSON()
  if (Array.isArray(json)) {
    return json[0]
  }

  return json
}

/**
 * @param setupCode imports, common code
 * @example
 * const expectSnippet = expecter(`
 *  import { add } from './made-up/add'
 * `);
 *
 * expectSnippet(`
 *   const x = add(10, 20)
 *