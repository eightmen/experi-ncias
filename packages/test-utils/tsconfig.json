{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "target": "ES2018"
  },
  "include": ["src/**/*.ts", "src/**/*.tsx", "src/**/*.d.ts"]
}
