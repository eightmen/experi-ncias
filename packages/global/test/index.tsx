/**
 * @jest-environment jsdom
 * @jsx jsx
 */

import { jsx } from '@theme-ui/core'
import { cleanup } from '@testing-library/react'
import { render } from '@theme-ui/test-utils'
import { matchers } from '@emotion/jest'

import { ThemeProvider } from '@theme-ui/core'
import Global from '../src'

expect.extend(matchers)

beforeEach(() => {
  document.head.innerHTML = ''
  jest.resetAllMocks()
})

afterEach(cleanup)

test.only('renders global styles', async () => {
  const root = (
    <ThemeProvider
      theme={{
        config: {
          useRootStyles: false,
        },
        fonts: {
          body: 'Georgia,serif',
        },
        colors: {
          primary: 'salmon',
        },
      }}
    >
      <Global
        styles={{
          '@font-face': {
            fontFamily: 'some-name',
          },
          body: {
            fontFamily: 'body',
            margin: 0,
          },
          h1: {
            color: 'primary',
          },
        }}
      />
      <h1>Hello</h1>
    </ThemeProvider>
  )

  const document = render(root)
  expect(document.baseElement.parentElement).toMatchSnapshot()

  const bodyStyle = global.getComputedStyle(document.baseElement)
  expect(bodyStyle.fontFamily).toBe('Georgia,serif')
  expect(bodyStyle.margin).toBe('0px')

  const h1 = document.baseElement.querySelector('h1')!
  expect(global.getComputedStyle(h1).color).toBe('salmon')
})

test('renders global and component styles', () => {
  const root = (
    <header>
      <Global
        styles={{
          html: {
            backgroundColor: 'hotpink',
          },
        }}
      />
      <div sx={{ color: 'pink' }} />
    </header>
  )
  const { baseElement } = render(root)
  expect(baseElement.parentElement).toMatchSnapshot()
})
