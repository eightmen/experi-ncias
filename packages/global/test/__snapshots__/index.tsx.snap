
// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders global and component styles 1`] = `
.emotion-0 {
  color: pink;
}

<html>
  <head>
    <style
      data-emotion="css"
      data-s=""
    >
      
      .css-2cx43c{}
    </style>
    <style
      data-emotion="css"
      data-s=""
    >
      
      .css-2cx43c html{background-color:hotpink;}
    </style>
    <style
      data-emotion="css"
      data-s=""
    >
      
      .emotion-0{color:pink;}
    </style>
  </head>
  <body>
    <div>
      <header>
        <div
          class="emotion-0"
        />
      </header>
    </div>
  </body>
</html>
`;

exports[`renders global styles 1`] = `
<html>
  <head>
    <style
      data-emotion="css-global"
      data-s=""
    >
      
      @font-face{font-family:some-name;}
    </style>
    <style
      data-emotion="css-global"
      data-s=""
    >
      
      body{font-family:Georgia,serif;margin:0;}
    </style>
    <style
      data-emotion="css-global"
      data-s=""
    >
      
      h1{color:salmon;}
    </style>
  </head>
  <body>
    <div>
      <h1>
        Hello
      </h1>
    </div>
  </body>
</html>
`;